### PR TITLE
feat(ui/message): support for multi images, collapse long text on messages

### LINKED ISSUE #373: [Web]: Adjust how many images can be shown per post preview
### What is this bug report for?

website, ui

### Issue description

When a post is made on discord and multiple images are uploaded with the post and then indexed, it will show ALL the images in a row prior to showing the next post

screencapture-chrome-extension-fdpohaocaechififmbbbbbknoalclacl-capture-html-2023-04-24-18_48_01

If you go to the link below and look at the 2nd post, you will see the issue I am referring to.
https://www.answeroverflow.com/c/1068915969789009970

### Code sample

_No response_

### Operating system

windows 10

### Priority this issue should have

Medium (should be fixed soon)

### Extra

_No response_

### PR DESCRIPTION
**Pull request by JoltCode**
# Description

- Support for multi images on a message (grid + modal)
- Collapses long strings on messages

Fixes #373 (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

N/A

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works 
- [x] I have updated .env.example if I added a new environment variable
- [x] My PR title follows the [semantic commits](https://www.conventionalcommits.org/en/v1.0.0/) style

### CODE REVIEW

**Comment by JoltCode** [comment]
Awaiting CI and then should be ready for review :+1:

File: packages/ui/src/components/primitives/Message.tsx [hunk 1]
```diff
@@ -7,17 +7,28 @@ import { DiscordAvatar } from './DiscordAvatar';
...
+import { useEffect } from 'react';
+import Lightbox, { type Slide } from 'yet-another-react-lightbox';
+import { Zoom, Counter } from 'yet-another-react-lightbox/plugins';
+import 'yet-another-react-lightbox/styles.css';
+import { ChevronDownIcon, ChevronUpIcon } from '@heroicons/react/24/outline';
 
 // eslint-disable-next-line @typescript-eslint/naming-convention
 const MessageContext = createContext<{
 	message: APIMessageWithDiscordAccount;
+	collapseContent?: boolean;
```
Comment (RhysSullivan) [inline packages/ui/src/components/primitives/Message.tsx:31]: This should be the same as the props that the top level message component takes in

Comment (RhysSullivan) [reply packages/ui/src/components/primitives/Message.tsx:31]: Extract to type and reuse i.e MessageProps

File: packages/ui/src/components/primitives/Message.tsx [hunk 2]
```diff
@@ -30,6 +41,70 @@ export function useMessageContext() {
 	return context;
 }
 
+const getImageHeightWidth = async ({ imageSrc }: { imageSrc: string }) => {
```
Comment (RhysSullivan) [inline packages/ui/src/components/primitives/Message.tsx:44]: dont put this in the message file, it makes it messy

File: packages/ui/src/components/primitives/Message.tsx [hunk 3]
```diff
@@ -60,99 +135,216 @@ export const MessageAuthorArea = () => {
 	);
 };
 
-export const MessageContents = () => {
+const LongMessageContents = () => {
 	const { message } = useMessageContext();
 	const { toHTML } = discordMarkdown;
 	const convertedMessageContent = toHTML(message.content);
+	const [messageContentToRender, setMessageContentToRender] = useState<string>(
```
Comment (RhysSullivan) [inline packages/ui/src/components/primitives/Message.tsx:142]: axe this

File: packages/ui/src/components/primitives/Message.tsx [hunk 4]
```diff
@@ -60,99 +135,216 @@ export const MessageAuthorArea = () => {
 	);
 };
 
-export const MessageContents = () => {
+const LongMessageContents = () => {
```
Comment (RhysSullivan) [inline packages/ui/src/components/primitives/Message.tsx:138]: Get rid of expand / collapse of message content

File: packages/ui/src/components/primitives/Message.tsx [hunk 5]
```diff
@@ -60,99 +135,216 @@ export const MessageAuthorArea = () => {
...
-}: React.PropsWithChildren<{
-	attachment: APIMessageWithDiscordAccount['attachments'][number];
-}>) => {
-	const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+export const MessageContents = () => {
+	const { message, collapseContent } = useMessageContext();
+	const { toHTML } = discordMarkdown;
+	const convertedMessageContent = toHTML(message.content);
+
+	if (message.content.length < 1000 || collapseContent === false)
```
Comment (RhysSullivan) [inline packages/ui/src/components/primitives/Message.tsx:188]: message.content.length < 1000

Pass in as an optional prop, max message length, with a default, maybe make it a bool | number, if it's set to true, use the default, if its set to a number use that

Comment (RhysSullivan) [reply packages/ui/src/components/primitives/Message.tsx:188]: Careful with implemnting this, a number will be typecast to true so make sure to check the type

**Review by RhysSullivan — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/primitives/Message.tsx (hunk 4 above)
Comment (JoltCode) [reply packages/ui/src/components/primitives/Message.tsx:138]: Done :+1:

File: packages/ui/src/components/primitives/Message.tsx (hunk 5 above)
Comment (JoltCode) [reply packages/ui/src/components/primitives/Message.tsx:188]: :+1:

File: packages/ui/src/components/pages/MessageResultPage.tsx [hunk 6]
```diff
@@ -91,6 +91,7 @@ export function MessageResultPage({
 				key={message.id}
 				message={message}
 				fullRounded
+				collapseContent={false}
```
[the remark this answers is not in the document]
Comment (JoltCode) [reply packages/ui/src/components/pages/MessageResultPage.tsx:94]: :+1:

File: packages/ui/src/components/primitives/Message.tsx (hunk 1 above)
Comment (JoltCode) [reply packages/ui/src/components/primitives/Message.tsx:31]: Fixed :+1: - hope that's what you meant :)

File: packages/ui/src/components/primitives/Message.tsx [hunk 7]
```diff
@@ -60,99 +120,202 @@ export const MessageAuthorArea = () => {
...
+		return (
+			<div className="flex h-[50vh] items-center justify-center">
+				<div className="h-32 w-32 animate-spin rounded-full border-b-4 border-ao-blue" />
+			</div>
+		);
+	}
+
+	if (parsedImages === 'error') {
+		return (
+			<Heading.H2 className="mt-2 text-lg">
```
Comment (RhysSullivan) [inline packages/ui/src/components/primitives/Message.tsx:190]: this should not be an h2 make it a p

**Review by RhysSullivan — CHANGES_REQUESTED** [review]
needs a storybook story for search results and message result pages showing that collapsing is working on them. looks like youre missing the collapse prop on those

File: packages/ui/src/components/primitives/Message.stories.tsx [hunk 8]
```diff
@@ -12,6 +12,16 @@ const meta = {
 			disable: true,
 		},
 	},
+	argTypes: {
+		collapseContent: {
+			control: {
+				type: 'boolean',
```
[the remark this answers is not in the document]
Comment (JoltCode) [reply packages/ui/src/components/primitives/Message.stories.tsx:18]: Fixed :+1:

File: packages/ui/src/components/primitives/Message.tsx (hunk 7 above)
Comment (JoltCode) [reply packages/ui/src/components/primitives/Message.tsx]: Fixed :+1:

File: packages/ui/src/components/primitives/Message.stories.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (JoltCode) [reply packages/ui/src/components/primitives/Message.stories.tsx]: Done :+1:

File: packages/ui/src/components/pages/MessageResultPage.tsx [hunk 9]
```diff
@@ -91,6 +91,7 @@ export function MessageResultPage({
 				key={message.id}
 				message={message}
 				fullRounded
+				collapseContent
```
[the remark this answers is not in the document]
Comment (JoltCode) [reply packages/ui/src/components/pages/MessageResultPage.tsx:94]: Fixed :+1:

File: packages/ui/src/components/pages/SearchPage.stories.tsx [hunk 10]
```diff
@@ -58,6 +58,46 @@ export const Results: Story = {
...
 	},
 };
+
+export const ResultsWithSuperLongMessage: Story = {
+	args: {
+		results: [
+			{
+				message: {
+					...mockMessageWithDiscordAccount({
+						content: `Lorem ipsum, dolor sit amet consectetur adipisicing elit. Molestias, et possimus, fuga est itaque impedit laborum labore similique tempora quas atque minus voluptatum explicabo blanditiis magni obcaecati! Error, deleniti id.
```
[the remark this answers is not in the document]
Comment (JoltCode) [reply packages/ui/src/components/pages/SearchPage.stories.tsx]: Fixed :+1:

**Pull request merged**